### PR TITLE
Updated Platforms/JVM section structure

### PR DESCRIPTION
**Pull request by danil-pavlov**
- update: prepared new toc structure for the jvm section
- update: added links to spring documentation and tutorials
- https://youtrack.jetbrains.com/issue/KT-45208

### CODE REVIEW

File: docs/topics/jvm/jvm-spring-boot-restful.md [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-
-To get started, first download and install the latest version of [IntelliJ IDEA](https://www.jetbrains.com/idea/download/index.html).
-
-You can also watch a video of this tutorial:
+You can download and explore the [completed project](https://github.com/kotlin-hands-on/spring-time-in-kotlin-episode1)
+or check out our video for this tutorial:
 
 <video width="560" height="315" href="gf-kjD2ZmZk" title="Spring Time in Kotlin. Getting Started"/>
 
+To get started, first download and install the latest version of [IntelliJ IDEA](https://www.jetbrains.com/idea/download/index.html).
```
[the remark this answers is not in the document]
Comment (koshachy) [reply docs/topics/jvm/jvm-spring-boot-restful.md:17]: This line is almost invisible after the video block (see staging) but important.
So maybe it's better to create a separate section like "Prerequisites" or "Before you start"?

File: docs/kr.tree [hunk 2]
```diff
@@ -119,15 +119,19 @@
...
+                <toc-element id="jvm-spring-boot-restful.md" accepts-web-file-names="spring-boot-restful.html,spring-boot-restful-db.html,jvm-spring-boot-restful-db.html"/>
+                <toc-element toc-title="Spring Framework Documentation for Kotlin" href="https://docs.spring.io/spring-framework/docs/current/reference/html/languages.html#languages"/>
+                <toc-element toc-title="Build a web application with Spring Boot and Kotlin – tutorial" href="https://spring.io/guides/tutorials/spring-boot-kotlin/"/>
+                <toc-element toc-title="Create a chat application with Kotlin Coroutines and RSocket – tutorial" href="https://spring.io/guides/tutorials/spring-webflux-kotlin-rsocket/"/>
+            </toc-element>
+            <toc-element id="jvm-test-using-junit.md"/>
             <toc-element id="mixing-java-kotlin-intellij.md"/>
-            <toc-element id="comparison-to-java.md"/>
             <toc-element id="java-to-kotlin-idioms-strings.md"/>
+            <toc-element id="jvm-records.md"/>
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/kr.tree:134]: The reason was the GA statistics: Strings section is 6 times more popular than Records. But maybe the stats is not that important in this case, what do you think?

File: docs/topics/jvm/jvm-spring-boot-restful.md [hunk 3]
```diff
@@ -139,7 +148,181 @@ The application is now ready to run:
 
    ![Application output](spring-boot-output.png)
 
-## Proceed to the next tutorial
+## Add database support
 
-In the next part of this tutorial you will add a database for storing objects, as well as two endpoints for writing and retrieving those objects: [Add a database to a Spring Boot RESTful web service](jvm-spring-boot-restful-db.md).
+In this section, you will create two endpoints: one for saving messages to the database, and one for retrieving them:
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/jvm/jvm-spring-boot-restful.md]: You're right. Rephrased the sentence

File: docs/topics/jvm/jvm-spring-boot-restful.md [hunk 4]
```diff
@@ -1,19 +1,21 @@
-[//]: # (title: Create a RESTful web service with Spring Boot – tutorial)
+[//]: # (title: Create a RESTful web service with a database using Spring Boot – tutorial)
 
-This tutorial walks you through the process of creating a simple application with Spring Boot.
+This tutorial walks you through the process of creating a simple application with Spring Boot and adding a database to
+a RESTful web service.
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/jvm/jvm-spring-boot-restful.md]: I agree. Rephrased the sentence

File: docs/topics/jvm/mixing-java-kotlin-intellij.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-[//]: # (title: Mixing Java and Kotlin in one project – tutorial)
+[//]: # (title: Mixing Java and Kotlin in a Java project – tutorial)
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/jvm/mixing-java-kotlin-intellij.md]: True, my mistake. Returned the old title

File: docs/topics/jvm/jvm-spring-boot-restful.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/jvm/jvm-spring-boot-restful.md]: Added a new section Before you start

File: docs/topics/jvm/jvm-spring-boot-restful.md [hunk 6]
```diff
@@ -1,19 +1,21 @@
...
 
-This tutorial consists of two parts:
-* Create a RESTful Web Service with Spring Boot
-* [Add a database to a Spring Boot RESTful web service](jvm-spring-boot-restful-db.md) 
-
-To get started, first download and install the latest version of [IntelliJ IDEA](https://www.jetbrains.com/idea/download/index.html).
-
-You can also watch a video of this tutorial:
+You can download and explore the [completed project](https://github.com/kotlin-hands-on/spring-time-in-kotlin-episode1)
+or check out our video for this tutorial:
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/jvm/jvm-spring-boot-restful.md]: Reversed to the previous wording

File: pages/docs/tutorials/spring-boot-restful-db.md [hunk 7]
```diff
@@ -4,4 +4,4 @@ showAuthorInfo: false
 redirect_path: https://kotlinlang.org/docs/jvm-spring-boot-restful-db.html
 ---
 
-The page is moved to [topics/jvm/jvm-spring-boot-restful-db.md](../../../docs/topics/jvm/jvm-spring-boot-restful-db.md)
+The page is moved to [topics/jvm/jvm-spring-boot-restful.md](../../../docs/topics/jvm/jvm-spring-boot-restful.md)
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply pages/docs/tutorials/spring-boot-restful-db.md]: Removed the file. @zoobestik confirmed that we don't need these files for WH

File: docs/topics/jvm/jvm-spring-boot-restful.md [hunk 8]
```diff
@@ -139,7 +148,181 @@ The application is now ready to run:
...
+
+```bash
+curl -X POST --location "http://localhost:8080" -H "Content-Type: application/json" -d "{ \"text\": \"Hello!\" }"
+
+curl -X POST --location "http://localhost:8080" -H "Content-Type: application/json" -d "{ \"text\": \"Bonjour!\" }"
+
+curl -X POST --location "http://localhost:8080" -H "Content-Type: application/json" -d "{ \"text\": \"Privet!\" }"
+
+curl -X GET --location "http://localhost:8080"
+```
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/jvm/jvm-spring-boot-restful.md:330]: Done, added links to Spring tutorials

File: docs/kr.tree (hunk 2 above)
[the remark this answers is not in the document]
Comment (p7nov) [reply docs/kr.tree:134]: I expect that we'll have more pages like the string idioms guide, so having them together seems right. If we add more such pages and group them, having a single unconnected page Records after them will look weird.

[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/kr.tree:133]: Got it. Swapped these lines

[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/kr.tree:133]: Have you seen that Anton Arhipov suggested (in Slack) to place Java Records and migration guides above Spring? I suppose that he wanted to do so to have all interop pages together.

[the remark this answers is not in the document]
Comment (PetrakovichVictoria) [reply docs/kr.tree:133]: Discussed in DM.

**Review by PetrakovichVictoria — APPROVED** [review]
I approve the new structure.

**Review by koshachy — APPROVED** [review]

**Review by p7nov — APPROVED** [review]
👍

**Pull request merged**